src/services/blockchain/ethereum/ethereum-blockchain-service.ts

```diff
@@ -109,35 +128,31 @@ export default class EthereumBlockchainService implements BlockchainService {
    * previous attempt's gas, even if the gas price on chain has gone down since then. This is because
    * retries of the same transaction (using the same nonce) need to have a gas price at least 10% higher
    * than any previous attempts or the transaction will fail.
-   * @param currentGasEstimate
+   * @param feeData
    * @param attempt
    * @param previousGas
    */
   static increaseGasPricePerAttempt(
-    currentGasEstimate: BigNumber,
+    feeData: providers.FeeData,
     attempt: number,
     previousGas: BigNumberish | undefined
   ): BigNumber {
-    const tenPercent = currentGasEstimate.div(10)
-    const additionalGas = tenPercent.mul(attempt)
-    const newGas = currentGasEstimate.add(additionalGas)
+    // Ganache still does not support 1559, so here is a workaround
+    const maxPriorityFeePerGas = feeData.maxPriorityFeePerGas || feeData.gasPrice
```

Comment:
Why is it or gasprice? Is this the workaround? If so it seems better to have two helper methods for the calculation. One that is like legacy and one that is 1559